catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/ResourceCacheImpl.java

```diff
@@ -109,15 +108,12 @@ private Config getHazelcastConfig(BundleContext context, String xmlConfigFilenam
         cfg = xmlConfigBuilder.build();
         LOGGER.debug(
             "Successfully built hazelcast config from XML config file {}", xmlConfigFilename);
-      } catch (FileNotFoundException e) {
-        LOGGER.info(
-            "FileNotFoundException trying to build hazelcast config from XML file "
-                + xmlConfigFilename,
-            e);
-        cfg = null;
       } catch (IOException e) {
-        LOGGER.info(
-            "IOException trying to build hazelcast config from XML file " + xmlConfigFilename, e);
+        StringBuilder info = new StringBuilder();
+        info.append(e.getClass().getSimpleName());
+        info.append(" trying to build hazelcast config from XML file ");
+        info.append(xmlConfigFilename);
+        LOGGER.info(info.toString());
```

Comment:
nit : instead of using StringBuilder, couldn't you do something like : `LOGGER.info("{} trying to build hazelcast config from XML file {}", e.getClass().getSimpleName(), xmlConfigFilename);`?

Reply:
can do

Reply:
Check the following comment first as you removed the tracing of the exception and if you use formatting, that will not work for tracing exceptions the way it was done

Reply:
i believe the example above will log the exception

Reply:
No it does not. The last argument (i.e.) is treated as an argument to the format string and not as an exception argument to trace along with its stack trace

Reply:
Sample output from the proposed logging statement : ```11:50:54,301 | TRACE  | tp1697072560-270 | pensearch.endpoint.OpenSearchEndpoint  138 | talog-opensearch | IOException trying to build hazelcast config from XML file test
java.io.IOException: IOException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:180) ~[132:org.apache.cxf.cxf-core:3.1.11]
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) ~[132:org.apache.cxf.cxf-core:3.1.11]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:189) ~[145:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.11]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:99) ~[145:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.11]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) ~[132:org.apache.cxf.cxf-core:3.1.11]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) ~[132:org.apache.cxf.cxf-core:3.1.11]
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308) ~[132:org.apache.cxf.cxf-core:3.1.11]
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) ~[132:org.apache.cxf.cxf-core:3.1.11]
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:262) ~[158:org.apache.cxf.cxf-rt-transports-http:3.1.11]
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) ~[158:org.apache.cxf.cxf-rt-transports-http:3.1.11]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) ~[158:org.apache.cxf.cxf-rt-transports-http:3.1.11]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) ~[158:org.apache.cxf.cxf-rt-transports-http:3.1.11]
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:180) ~[158:org.apache.cxf.cxf-rt-transports-http:3.1.11]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:299) ~[158:org.apache.cxf.cxf-rt-transports-http:3.1.11]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doGet(AbstractHTTPServlet.java:223) ~[158:org.apache.cxf.cxf-rt-transports-http:3.1.11]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687) ~[11:javax.servlet-api:3.1.0]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:274) ~[158:org.apache.cxf.cxf-rt-transports-http:3.1.11]```

Reply:
You shouldn't have to get the exception classname like that. Simply do:
```LOGGER.info("Failed to build hazelcast config from XML file " + xmlConfigFilename, e)```
With this you get the stack trace of the exception and the exception toString() logged automatically by the logging API. Make sure you do not use a format message with additional arguments though as it needs to call the overloaded method info(String, Throwable)